File: src/lib/Mate.java

package lib;

public class Mate {
	
	public int Potencia(int base, int exponente) {
		int pot=1;
		for (byte i=1;i<=exponente;i++)
			pot*=base;
		return pot;
	}
	
	public int Factorial(int num) {
		int fact=1;
		for (byte i=1;i<=num;i++)
			fact*=i;
		return fact;
	}
	
	public int Absoluto(int num) {
		if (num<0) num*=-1;
		return num;
	}
	
}

File: Mate.pas

unit Mate;

interface

function Potencia(base,exponente:cardinal):cardinal;
function Factorial(num:byte):cardinal;
function Absoluto(num:integer):cardinal;


implementation

// 2^4 --> 16
function Potencia(base,exponente:cardinal):cardinal;
var
  pot:cardinal;
  i:byte;
begin
  pot:=1;
  for i := 1 to exponente do
    pot:=pot*base;
  Potencia:=pot;
end;

// 3! --> 6
function Factorial(num:byte):cardinal;
var
  fact:cardinal;
  i:byte;
begin
  fact:=1;
  for i := 1 to num do
    fact:=fact*i;
  Factorial:=fact;
end;

// |-9 | --> 9
function Absoluto(num:integer):cardinal;
begin
  if num<0 then
    Absoluto:=num*-1
  else
    Absoluto:=num;
end;


end.
